### DISCUSSION TITLE
How to set chromeWebSecurity for a single test?

### DISCUSSION THREAD

**Discussion by debasisj** [cypress-io/cypress#29890] [Questions and Help]
Cypress.config("chromeWebSecurity", false) -> this does not work anymore!! as compilation error `Argument of type '"chromeWebSecurity"' is not assignable to parameter of type 'keyof TestConfigOverrides'.ts(2345)`

and also the option not available for `cy.visit(url, config,fn)` level as well? 

I have this credit card iframe that is from cross origin. However setting chromeWebSecurity as false at default config fails some other test.

Cypress version `13.3.3`
Any suggestion?

**Comment by portyu9** [comment]
Hi @debasisj 

`chromeWebSecurity` cannot be changed for a single `it()` or dynamically with `Cypress.config()`.

It is a **browser-launch configuration option**, so by the time the test is running it is too late to change it. That is why TypeScript does not include it in `TestConfigOverrides`.

For your payment/credit-card iframe spec, the cleanest isolation is to run that spec separately with web security disabled:

```sh
npx cypress run \
  --browser chrome \
  --spec "cypress/e2e/payment.cy.ts" \
  --config chromeWebSecurity=false
```

Then run the rest of the suite normally:

```sh
npx cypress run --browser chrome
```

Another option is a separate config file for that group of specs:

```sh
npx cypress run \
  --config-file cypress.payment.config.ts \
  --spec "cypress/e2e/payment.cy.ts"
```

with:

```ts
import { defineConfig } from 'cypress'

export default defineConfig({
  chromeWebSecurity: false,
  e2e: {
    // ...
  },
})
```

A few important limitations:

- disabling web security only works with Chromium-family browsers;
- `cy.origin()` does **not** provide access to a cross-origin iframe — it is for top-level cross-origin navigation;
- if the payment provider offers a test/sandbox API or test tokenization path, that is generally preferable to coupling the suite to the provider's real hosted iframe UI.

So there is no supported per-test toggle; use a separate Cypress invocation/configuration for the specs that require it.

No screenshots or attachments were included in this discussion.

Please mark this answer as accepted if it helped, thank you.
